<?xml version="1.0"?>
<launch>
    <arg name="path_to_orb" default="/home/usrg/packages/ORB_SLAM3_ROS_Interface/"/>
    <arg name="path_to_vocabulary" default="Vocabulary/ORBvoc.txt"/>
    <!-- <arg name="path_to_settings" default="Examples/RGB-D/RealSense_BagPlay.yaml"/> -->
    <arg name="path_to_settings" default="Examples/orb3_ros_interface/config/RealSense_D435i.yaml"/>

    <arg name="use_pangolin" default="false"/>
    <arg name="use_auto_exposure" default="true"/>
    <arg name="ae_meanIntensitySetPoint" default="1500"/>

    <!-- conversion node -->
    <node name="rgbd_sub_topic" pkg="orb3_ros_interface" type="rgbd_sub_topic" output="screen"
    args="$(arg path_to_orb)$(arg path_to_vocabulary) $(arg path_to_orb)$(arg path_to_settings)" >

      <remap from="/orb_pose" to="/scout/mavros/vision_pose/pose"/>
      <remap from="/camera/depth/image_rect_raw" to="/camera/aligned_depth_to_color/image_raw"/>

      <param name="enable_pangolin" type="bool" value="$(arg use_pangolin)" />
      <param name="enable_auto_exposure" type="bool" value="$(arg use_auto_exposure)" />
      <param name="ae_meanIntensitySetPoint" type="int" value="$(arg ae_meanIntensitySetPoint)" />

      <remap from="/camera/color/image_raw" to="/camera/image_conversed"/> 
      <remap from="/orb_odom" to="/orb_odom_conversion"/>
      <remap from="/ORB3/globalmap" to="/ORB3_Conversion/globalmap"/> 
      <remap from="/ORB3/localmap"  to="/ORB3_Conversion/localmap"/> 

      <!-- <remap from="/orb3_feature_image_from_orb" to="/orb3_feature_image_processed"/> -->

      
      
    </node>


    <!-- originanl node -->
    
    <node name="rgbd_sub_topic_original" pkg="orb3_ros_interface" type="rgbd_sub_topic" output="screen"
    args="$(arg path_to_orb)$(arg path_to_vocabulary) $(arg path_to_orb)$(arg path_to_settings)" >
      <remap from="/orb_pose" to="/scout/mavros/vision_pose/pose_original"/>
      <remap from="/camera/depth/image_rect_raw" to="/camera/aligned_depth_to_color/image_raw"/>

      <param name="enable_pangolin" type="bool" value="$(arg use_pangolin)" />
      <param name="enable_auto_exposure" type="bool" value="$(arg use_auto_exposure)" />
      <param name="ae_meanIntensitySetPoint" type="int" value="$(arg ae_meanIntensitySetPoint)" />

      <remap from="/orb3_feature_image_from_orb" to="/orb3_feature_image_original"/> 
    </node>
   


</launch>
